### PR TITLE
Raspberrypicode

### PR DESCRIPTION
**Pull request by Mikhail-Kreytser**

### CODE REVIEW

File: openSpotServer/carDetection/carDetectionModel.py [hunk 1]
```diff
@@ -75,80 +76,57 @@ def find_cars(image_path):
...
-    h = 3000
-    w = 3000
-    crop = rgb[y:y + h, x:x + w]
+    crop = rgb[crop_y:crop_y+crop_h, crop_x:crop_x+crop_w]
 
     # Run the image through the Mask R-CNN model to get results.
     results = model.detect([crop], verbose=0)
 
     # Mask R-CNN assumes we are running detection on multiple images.
     # We only passed in one image to detect, so only grab the first result.
```
Comment (khelif96) [inline openSpotServer/carDetection/carDetectionModel.py]: We can prob start getting rid of these comments

File: openSpotServer/carDetection/carDetectionModel.py [hunk 2]
```diff
@@ -75,80 +76,57 @@ def find_cars(image_path):
...
+    red=[0,0,1]
+    green=[0,1,0]
+    new_image = crop
     for i in fullSpots:
-        new_image = visualize.apply_mask(crop, i,color1, alpha=1)
+        new_image = visualize.apply_mask(crop, i,red, alpha=0.5)
 
     for i in emptySpots:
-        new_image = visualize.apply_mask(new_image, i,color2, alpha=1)  
+        new_image = visualize.apply_mask(new_image, i,green, alpha=0.5)
```
Comment (khelif96) [inline openSpotServer/carDetection/carDetectionModel.py]: Noice

File: openSpotServer/core/cropSettings.py [hunk 3]
```diff
@@ -0,0 +1,4 @@
+crop_x = 0
+crop_y = 450
+crop_h = 1350
+crop_w = 2592
```
Comment (khelif96) [inline openSpotServer/core/cropSettings.py]: EOL

File: openSpotServer/core/views.py [hunk 4]
```diff
@@ -17,9 +17,16 @@ def home(request):
 
 
 def custom_save(image_form):
```
Comment (khelif96) [inline openSpotServer/core/views.py]: This function name is a little confusing. What are we doing just deleting the old image and saving the new one?

File: openSpotServer/parkingSpot/parkingSpots.py [hunk 5]
```diff
@@ -1,9 +1,23 @@
...
+spot_19 = np.array([[430,605], 	[655,610], 	[525,672],	  [315,660]], np.int32)
+spot_20 = np.array([[220,595], 	[390,600], 	[285,657],	  [95,655]], np.int32)
+spot_21 = np.array([[0,590], 	[165,595], 	[60,650],	  [0,646]], np.int32)
 
-spot_15 = np.array([[1465,152], [1700,152], [1700,248],   [1430,248]], np.int32)
-spot_16 = np.array([[1210,152], [1430,152], [1400,243],   [1135,243]], np.int32)
-spot_17 = np.array([[950,152],  [1175,152], [1100,240],   [830,240]], np.int32)
-spot_list=[spot_15,spot_16,spot_17]
\ No newline at end of file
+spot_list = [ spot_13, spot_14, spot_15, spot_16, spot_17, spot_18, spot_19, spot_20, spot_21]
```
Comment (khelif96) [inline openSpotServer/parkingSpot/parkingSpots.py]: Are these all the spots?

File: openSpotServer/parkingSpot/parkingSpots.py [hunk 6]
```diff
@@ -1,9 +1,23 @@
...
 #                     Top Left, Top Right,  Bottom Right, Bottom Left
+spot_13 = np.array([[1693,620], [1900,620], [1962,698],   [1736,698]], np.int32)
+spot_14 = np.array([[1485,620], [1675,620], [1710,698],   [1490,696]], np.int32)
+spot_15 = np.array([[1282,620], [1469,620], [1470,695],   [1250,693]], np.int32)
+spot_16 = np.array([[1080,620], [1265,620], [1230,692],   [1015,690]], np.int32)
+spot_17 = np.array([[860,620], 	[1063,620], [995,690],	  [778,686]], np.int32)
+spot_18 = np.array([[669,610], 	[839,617], 	[760,682],	  [550,675]], np.int32)
+spot_19 = np.array([[430,605], 	[655,610], 	[525,672],	  [315,660]], np.int32)
+spot_20 = np.array([[220,595], 	[390,600], 	[285,657],	  [95,655]], np.int32)
+spot_21 = np.array([[0,590], 	[165,595], 	[60,650],	  [0,646]], np.int32)
```
Comment (khelif96) [inline openSpotServer/parkingSpot/parkingSpots.py]: Nitpick but could you fix the spacing

File: raspberryPi/raspberryPi.py [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+                data={"csrfmiddlewaretoken": csrf_token, "description": static_camera_identifier},
+                files={"image": image},
+            )
+
+
+print("Setting up the camera")
+# Light Mode 
+# camera = PiCamera()
+# camera.resolution = (2592, 1944)
+# camera.framerate = 15
```
Comment (khelif96) [inline raspberryPi/raspberryPi.py]: Does this mean the camera wont take a properly exposed picture during the day? if so could we update this so it toggles between day mode and night mode based off the time?

File: openSpotServer/carDetection/carDetectionModel.py (hunk 1 above)
Comment (Mikhail-Kreytser) [reply openSpotServer/carDetection/carDetectionModel.py]: ill leave them in to help people understands the code

File: openSpotServer/parkingSpot/parkingSpots.py (hunk 5 above)
Comment (Mikhail-Kreytser) [reply openSpotServer/parkingSpot/parkingSpots.py]: no just top row

File: openSpotServer/parkingSpot/parkingSpots.py (hunk 6 above)
Comment (Mikhail-Kreytser) [reply openSpotServer/parkingSpot/parkingSpots.py]: i have spacing like that to help visualize the Top Left, Top Right,  Bottom Right, Bottom Left pixel

File: openSpotServer/core/views.py (hunk 4 above)
Comment (Mikhail-Kreytser) [reply openSpotServer/core/views.py]: changed name

File: raspberryPi/raspberryPi.py (hunk 7 above)
Comment (Mikhail-Kreytser) [reply raspberryPi/raspberryPi.py]: will do later

File: openSpotServer/parkingSpot/parkingSpots.py (hunk 6 above)
Comment (khelif96) [reply openSpotServer/parkingSpot/parkingSpots.py]: Can you explain what you mean?

Comment (Mikhail-Kreytser) [reply openSpotServer/parkingSpot/parkingSpots.py]: image

**Pull request merged**